pkgs/applications/system/zxfer/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, bash
+, fetchFromGitHub
+, installShellFiles
+, makeWrapper
+, coreutils
+, gawk
+, gnugrep
+, gnused
+, openssh
+, resholve
+, rsync
+, zfs
+}:
+
+resholve.mkDerivation rec {
+  pname = "zxfer";
+  version = "1.1.7";
+
+  src = fetchFromGitHub {
+    owner = "allanjude";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-11SQJcD3GqPYBIgaycyKkc62/diVKPuuj2Or97j+NZY=";
+  };
+
+  buildPhase = ''
+    substituteInPlace zxfer \
+      --replace 'AWK=$( which awk )'     'AWK=${gawk}/bin/awk' \
+      --replace 'AWK=$( which gawk )'    'AWK=${gawk}/bin/awk' \
+      --replace 'LCAT=""'                'LCAT=${coreutils}/bin/cat' \
+      --replace 'LZFS=$( which zfs )'    'LZFS=${zfs}/bin/zfs' \
+      --replace 'RSYNC=$( which rsync )' 'RSYNC=${rsync}/bin/rsync'
+  '';
+
+  nativeBuildInputs = [ installShellFiles makeWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    installManPage zxfer.1m zxfer.8
+    install -Dm755 zxfer -t $out/bin/
+
+    runHook postInstall
+  '';
+
+  solutions.default = {
+    scripts = [ "bin/zxfer" ];
+    interpreter = "${bash}/bin/bash";
+    inputs = [
+      coreutils
+      gnugrep
+      gnused
+      openssh
+    ];
+    fake.external = [
+      "kldload"
+      "kldstat"
+      "svcadm"
```

Comment:
Maybe good to note why these aren't available in a way that helps clarify if they're actionable later?
- Security wrappers? 
- Not packaged yet?
- Unpackageble? (proprietary system commands? etc.)

Reply:
yep, more investigation showed that they are builtins from other unixes. I have added comments to reflect their origin.

debian tried packaging kldutils, but seem to have given up, and I am unclear where opensolaris source lives.